### ISSUE TITLE
Can't find velocity in window and window.jQuery when using optimizer of RequireJS

### ISSUE THREAD

**Issue by PeachScript** [julianshapiro/velocity#486]
I use the following library by RequireJS. The Fragment of require config file:

```
...
paths: {
    'velocity': 'bower_components/velocity/velocity',
    'velocity.ui': 'bower_components/velocity/velocity.ui',
    'ngVelocity': 'bower_components/angular-velocity/angular-velocity',
}
...
```

And I got the following error in browser when I used the r.js optimize the javascript files:
```
Uncaught Velocity and Velocity UI Pack plugin required, please include the relevant JS files. Get Velocity with: bower install velocity
```

The exception be throw by the angular velocity. Then I check the file which be optimized, I found two code snippet:
**velocity.js**
```
    /* AMD module. */
    } else if (typeof define === "function" && define.amd) {
        define(factory);
    /* Browser globals. */
```
**velocity.ui.js**
```
    /* AMD module. */
    } else if (typeof define === "function" && define.amd) {
        define([ "velocity" ], factory);
    /* Browser globals. */
```

I try to edit the code to excuting the factory function, just like this`define([ "velocity" ], factory());`, then optimize them, the program running well !

I don not know whether the problem that I found is error or correct in the `velocity.js` and `velocity.ui.js`, so I write something that I know in this issue. I want to know the regular usage and the reason of my problem, thank you :D

**Comment by ydaniv** [comment]
This is either an angular-velocity issue or an issue with your use of RequireJS, not of Velocity.

**Comment by PeachScript** [comment]
Thanks for your reply ! I try to change the configuration of RequireJS  for a long time, but also error. Could you tell me how to use velocity and velocity.ui with RequireJS and angular-velocity?

This is my require config:
```
requirejs.config({
  baseUrl: '/',
  waitSeconds: 15,
  urlArgs: 'bust=' + new Date().getTime(),
  paths:{
    //Libraries
    'jQuery': 'bower_components/jquery/dist/jquery.min',
    'angular': 'bower_components/angular/angular',
    'ngRoute': 'bower_components/angular-route/angular-route.min',
    'ngCookies': 'bower_components/angular-cookies/angular-cookies.min',
    'ngAnimate': 'bower_components/angular-animate/angular-animate.min',
    'ui.bootstrap': 'bower_components/angular-bootstrap/ui-bootstrap-tpls.min',
    'velocity': 'bower_components/velocity/velocity.min',
    'velocity.ui': 'bower_components/velocity/velocity.ui.min',
    'ngVelocity': 'bower_components/angular-velocity/angular-velocity.min',

    //module, config, router, directive, service, controllers
    'app': 'js/app',
    'ab.config': 'js/config',
    'ab.route': 'js/route',
    'ab.service': 'js/service',
    'ab.directive': 'js/directive',
    'ab.controller': 'js/controller'
  },
  shim: {
    'velocity': {
      deps: ['jQuery']
    },
    'velocity.ui': {
      deps: ['velocity']
    },
    'angular': {
      deps: ['jQuery'],
      exports: 'angular'
    },
    'ngRoute': {
      deps: ['angular']
    },
    'ngCookies': {
      deps: ['angular']
    },
    'ngAnimate': {
      deps: ['angular']
    },
    'ui.bootstrap': {
      deps: ['angular']
    },
    'ngVelocity': {
      deps: ['velocity', 'velocity.ui', 'angular', 'ngAnimate']
    }
  }
});
```

And the code snippet of angular-velocity, it is not measure up AMD:
```
(function () {

	'use strict';

	var app = angular.module('angular-velocity', ['ngAnimate']);

	var CLASS_ANIM_ADD 	  = 0,
		CLASS_ANIM_REMOVE = 1,
		Velocity = window.Velocity || (window.jQuery && jQuery.Velocity);

	// Check we have velocity and the UI pack
	if (!Velocity || !Velocity.RegisterEffect) {
		throw "Velocity and Velocity UI Pack plugin required, please include the relevant JS files. Get Velocity with: bower install velocity";
	}

```

**Comment by ydaniv** [comment]
Sorry my friend, try on StackOverflow.

**Comment by PeachScript** [comment]
Got it, thank you very much.

**Comment by PeachScript** [comment]
Hi friend, I solve the problem just now. As you say, the problem be caused by me.

My configuration file is correct for RequireJS, but I use the r.js to compress code then I get the problem. In the docs of r.js, I found the configuration of `shim` has a option 'wrapShim', and the problem has been fixed when I set the `wrapShim` as true. The configuration file just like this:
```
requirejs.config({
  ...
  paths:{
    ...
  },
  shim: {
    ...
  },
  wrapShim: true
});
```
